features/core/src/main/java/org/apache/karaf/features/internal/service/Deployer.java

```diff
@@ -633,6 +633,9 @@ public void deploy(DeploymentState dstate, DeploymentRequest request) throws Exc
             toStop.addAll(regionDeployment.toUpdate.keySet());
             toStop.addAll(regionDeployment.toDelete);
         }
+        Set<Bundle> toRefreshToStopEarly = new HashSet<>(toRefresh.keySet());
+        toRefreshToStopEarly.remove(dstate.serviceBundle);
+        toStop.addAll(toRefreshToStopEarly);
```

Comment:
Is there a need for this intermediary collection ?
What about 
```
toStop.addAll(toRefresh.keySet());
toStop.remove(serviceBundle);
```

Reply:
intermediary was for case, when initial `toStop` contained service bundle. I didn't want to change that, as it worked.

Reply:
That's the part I don't get.
If it was supported, I don't see why you had a problem.  if it was not, we should not keep it.